File: app/src/main/java/rx_playground/com/jablonski/rxandroidplayground/submodels/SubmodelsViewContract.java
package rx_playground.com.jablonski.rxandroidplayground.submodels;

import rx_playground.com.jablonski.rxandroidplayground.mvp.BaseViewCotract;
import rx_playground.com.jablonski.rxandroidplayground.model.Model;

/**
 * Created by yabol on 23.04.2017.
 */

public interface SubmodelsViewContract {
    interface View extends BaseViewCotract.BaseView<Model> {
        void openModelDetails(String styleId);
    }

    interface Presenter extends BaseViewCotract.BasePresenter<Model>{
        void loadElements(String manufacturer, String niceName, String year);
    }

    interface Repository{
        void loadElements(String manufacturer, String modelNiceName, String year);
    }
}

File: app/src/main/java/rx_playground/com/jablonski/rxandroidplayground/submodels/SubmodelsViewContract.kt
package rx_playground.com.jablonski.rxandroidplayground.submodels

import io.reactivex.Observable
import rx_playground.com.jablonski.rxandroidplayground.model.Model
import rx_playground.com.jablonski.rxandroidplayground.model.ModelsResult
import rx_playground.com.jablonski.rxandroidplayground.mvp.BaseViewCotract

/**
 * Created by yabol on 23.04.2017.
 */

interface SubmodelsViewContract {
    interface View : BaseViewCotract.BaseView<Model> {
        fun openModelDetails(styleId: String)
    }

    interface Presenter : BaseViewCotract.BasePresenter<Model> {
        fun loadElements(manufacturer: String, niceName: String, year: String)
    }

    interface Repository {
        fun loadElements(manufacturer: String, modelNiceName: String, year: String): Observable<ModelsResult>
    }
}
